iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothubservices/errorinjection/SendMessagesErrInjTests.java

```diff
@@ -281,6 +281,68 @@ public void sendMessagesWithTcpConnectionDropNotifiesUserIfRetryExpires() throws
         testInstance.client.setRetryPolicy(new ExponentialBackoffWithJitter());
     }
 
+    @Test
+    public void clientCanReopenAfterDisconnectionEvent() throws Exception
+    {
+        if (testInstance.protocol == HTTPS || testInstance.useHttpProxy)
+        {
+            return;
+        }
+
+        this.testInstance.setup();
+
+        this.testInstance.client.setRetryPolicy(new NoRetry());
+
```

Comment:
Can you also add a test for retrying and recovering by explicitly setting lower retry timeout?

Reply:
So that the retry policy gets exhausted, and then application level retry is successful? Sure

Reply:
Actually, this kind of test isn't possible to write right now. The fault injection framework we use does not allow us to cause a retryable error for X number of seconds. That functionality is necessary for us to test running through the sdk level retry and some application level retry. We only get DISCONNECTED_RETRYING followed by a successful reconnect within the SDK retry logic